statistics/longest_standing_records.rb

```diff
@@ -50,10 +50,14 @@ def transform(query_results)
           end
           .group_by { |result| result["event_id"] }
           .flat_map do |event_id, results|
-            results.each_cons(2) do |previous_record, new_record|
-              previous_record["days"] = new_record["competition_date"] - previous_record["competition_date"]
-            end
-            results.last["days"] = Date.today - results.last["competition_date"]
+	    results.each do |result_1|
+	      result_1["days"] = Date.today - result_1["competition_date"]
+	      results.each do |result_2|
+		if result_2[type] < result_1[type]
+		  result_1["days"] = [result_1["days"], result_2["competition_date"] - result_1["competition_date"]].min
+		end
+	      end
+	    end
```

Comment:
I think this does the same but is a bit more explicit about the intent:

```suggestion
            results.each do |result_1|
              better_result = results.find { |result_2| result_2[type] < result_1[type] }
              better_date = better_result ? better_result["competition_date"] : Date.today
              result_1["days"] = better_date - result_1["competition_date"]
            end
```

(it could be optimised more, but I think in this case it's fine)

Reply:
Yes, this also works, thaks :)